modules/task/cherami/cherami.go

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cherami
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.uber.org/fx/modules/task"
+	"go.uber.org/fx/service"
+	"go.uber.org/fx/ulog"
+
+	"github.com/pkg/errors"
+	"github.com/uber-go/tally"
+	"github.com/uber/cherami-client-go/client/cherami"
+	cherami_gen "github.com/uber/cherami-thrift/.generated/go/cherami"
+	"go.uber.org/zap"
+)
+
+// TODO: Most of these will be fetched from host configuration instead of being hardcoded
+const (
+	_initialized = iota
+	_running
+	_stopped
+	_pathPrefix = "/uberfx_async/"
+)
+
+var (
+	_hyperbahnMu         sync.RWMutex
+	_hyperbahnHostsFile  string
+	_cheramiClientFunc   = cherami.NewHyperbahnClient
+	_defaultClientConfig = clientConfig{
+		ConsumerName:       "uberfx-async",
+		PrefetchCount:      10,
+		ConsumeWorkerCount: 10,
```

Comment:
PRs are not a good place for feature requests :) I think I mentioned this to you, if you can go ahead and cut a task for this, I will prioritize.